packages/flame/lib/src/experimental/follow_behavior.dart

```diff
@@ -0,0 +1,75 @@
+import '../components/component.dart';
+import '../components/position_component.dart';
+import '../effects/provider_interfaces.dart';
+import 'viewfinder.dart';
+import 'viewport.dart';
+
+/// This behavior will make the [owner] follow the [target].
+///
+/// Here, both the owner and the target are [PositionProvider]s, which could be
+/// either [PositionComponent]s, or camera's [Viewfinder]/[Viewport], or any
+/// other objects, including custom implementations.
+///
+/// The [maxSpeed] parameter controls the maximum speed with which the [owner]
+/// is allowed to move as it pursues the [target]. By default, the max speed is
+/// infinite, allowing the owner to stay on top of the target all the time.
+///
+/// The flags [horizontalOnly]/[verticalOnly] allow constraining the [owner]'s
+/// movement to the horizontal/vertical directions respectively.
+class FollowBehavior extends Component {
```

Comment:
You planned to add constraints to within which bounds the camera would move if I remember correctly, are you intending to put those bounds on this class or directly on the viewfinder, or maybe both?

Reply:
My plan was to have a separate component for that. Something like `ConstrainedPositionBehavior`. You would add it after the `FollowBehavior`, and it would check whether the position stays within the specified bounds (given by a Shape). If not, the camera would be moved towards the shape's boundary.